packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts

```diff
@@ -0,0 +1,122 @@
+ */
+export interface KafkaEventSourceProps extends StreamEventSourceProps {
+  /**
+   * the Kafka topic to subscribe to
+   */
+  readonly topic: string,
+  /**
+   * the secret with the Kafka credentials, see https://docs.aws.amazon.com/msk/latest/developerguide/msk-password.html for details
+   */
+  readonly secret: secretsmanager.ISecret
+}
+
+/**
+ * Properties for a MSK event source
+ */
+export interface ManagedKafkaEventSourceProps extends KafkaEventSourceProps {
+  /**
+   * the ARN of the MSK cluster
+   */
+  readonly clusterArn: string
+}
+
+/**
+ * Properties for a self managed Kafka cluster event source.
+ * If your Kafka cluster is only reachable via VPC make sure to configure it.
+ */
+export interface SelfManagedKafkaEventSourceProps extends KafkaEventSourceProps {
+  /**
+   * list of Kafka brokers
+   */
+  readonly bootstrapServers: string[]
+
+  /**
+   * If your Kafka brokers are only reachable via VPC, provide the subnets here
+   *
+   * @default - none
+   */
+  readonly subnets?: ISubnet[],
+
+  /**
+   * If your Kafka brokers are only reachable via VPC, provide the security group here
+   *
+   * @default - none
+   */
+  readonly securityGroup?: ISecurityGroup
+}
+
+/**
+ * Use a MSK cluster as a streaming source for AWS Lambda
+ */
+export class ManagedKafkaEventSource extends StreamEventSource<ManagedKafkaEventSourceProps> {
+
+  constructor(props: ManagedKafkaEventSourceProps) {
+    super(props);
+  }
+
+  public bind(target: lambda.IFunction) {
+    target.addEventSourceMapping(
+      `KafkaEventSource:${this.props.clusterArn.split('/').slice(-1)[0]}${this.props.topic}`,
+      this.enrichMappingOptions({
+        eventSourceArn: this.props.clusterArn,
+        startingPosition: this.props.startingPosition,
+        sourceAccessConfigurations: [{ type: 'SASL_SCRAM_512_AUTH', uri: this.props.secret.secretArn }],
+        kafkaTopic: this.props.topic,
+      }),
+    );
+
+    this.props.secret.grantRead(target);
+
+    target.addToRolePolicy(new iam.PolicyStatement(
+      {
+        actions: ['kafka:DescribeCluster', 'kafka:GetBootstrapBrokers', 'kafka:ListScramSecrets'],
+        resources: [this.props.clusterArn],
+      },
+    ));
+
+    target.role?.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaMSKExecutionRole'));
+  }
+}
+
+/**
+ * Use a self hosted Kafka installation as a streaming source for AWS Lambda.
+ */
+export class SelfManagedKafkaEventSource extends StreamEventSource<SelfManagedKafkaEventSourceProps> {
+
+  constructor(props: SelfManagedKafkaEventSourceProps) {
+    super(props);
+    if ((props.securityGroup !== undefined && props.subnets == undefined) || (props.securityGroup == undefined && props.subnets !== undefined )) {
+      throw new Error('both subnets and securityGroup must be set');
+    }
+  }
+
+  public bind(target: lambda.IFunction) {
+    let sourceAccessConfigurations = [{ type: 'SASL_SCRAM_512_AUTH', uri: this.props.secret.secretArn }];
```

Comment:
Looks like there are 3 possibilities here - `SASL_SCRAM_512_AUTH`, `SASL_SCRAM_256_AUTH` and `BASIC_AUTH`. We should model the type in the props.

Reply:
Good point. `BASIC_AUTH` is only supported by [ActiveMQ](https://docs.aws.amazon.com/lambda/latest/dg/with-mq.html#services-mq-api), not Kafka though.

Reply:
You can now choose.